### ISSUE TITLE
Quarto tries to source a non-existing conda environment

### ISSUE THREAD

**Issue by sorenwacker** [quarto-dev/quarto-cli#9271]
### Bug description

Suddenly, when doing `quarto preview` quarto fails with the error that the conda environment does not exist anymore. It was removed long time ago, and `quarto preview` was called several times in the mean time.

### Expected behavior

quarto should use the currently active environment.

### Actual behavior

It tries to open a different environment that was never created for quarto.

**Comment by sorenwacker** [comment]
```
quarto check
Quarto 1.4.552
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.552
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Tex:  (not detected)

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.2 (Conda)
      Path: /Users/sdrwacker/miniforge3/envs/jupyter/bin/python
      Jupyter: 5.7.2
      Kernels: ds, py3, python3

(-) Checking Jupyter engine render....2024-04-04 14:54:02,109 - traitlets - ERROR - Failed to run command:
['/Users/sdrwacker/miniconda3/envs/ds/bin/python', '-m', 'ipykernel_launcher', '-f', '/private/var/folders/hr/bq19zbjx0wvbr5gmvwypgb7431fw57/T/tmp8jkvskyr.json']
    PATH='.:/Applications/quarto/bin:/Users/sdrwacker/google-cloud-sdk/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/usr/local/laps:/Applications/quarto/bin:/Users/sdrwacker/miniforge3/envs/jupyter/bin:/Users/sdrwacker/miniconda3/condabin:/opt/homebrew/bin:/opt/homebrew/sbin:/scripts:/Users/sdrwacker/workspace/bash-config/scripts:/scripts:/Users/sdrwacker/workspace/bash-config/scripts'
    with kwargs:
{'stdin': -1, 'stdout': None, 'stderr': None, 'cwd': '/var/folders/hr/bq19zbjx0wvbr5gmvwypgb7431fw57/T/quarto-session84e73c42/6ecd65ab', 'start_new_session': True}

2024-04-04 14:54:02,109 - traitlets - ERROR - [Errno 2] No such file or directory: '/Users/sdrwacker/miniconda3/envs/ds/bin/python'
Traceback (most recent call last):
  File "/Users/sdrwacker/miniforge3/envs/jupyter/lib/python3.12/site-packages/jupyter_client/manager.py", line 87, in wrapper
    out = await method(self, *args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sdrwacker/miniforge3/envs/jupyter/lib/python3.12/site-packages/jupyter_client/manager.py", line 439, in _async_start_kernel
    await self._async_launch_kernel(kernel_cmd, **kw)
  File "/Users/sdrwacker/miniforge3/envs/jupyter/lib/python3.12/site-packages/jupyter_client/manager.py", line 354, in _async_launch_kernel
    connection_info = await self.provisioner.launch_kernel(kernel_cmd, **kw)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sdrwacker/miniforge3/envs/jupyter/lib/python3.12/site-packages/jupyter_client/provisioning/local_provisioner.py", line 210, in launch_kernel
    self.process = launch_kernel(cmd, **scrubbed_kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sdrwacker/miniforge3/envs/jupyter/lib/python3.12/site-packages/jupyter_client/launcher.py", line 170, in launch_kernel
    raise ex
  File "/Users/sdrwacker/miniforge3/envs/jupyter/lib/python3.12/site-packages/jupyter_client/launcher.py", line 155, in launch_kernel
    proc = Popen(cmd, **kwargs)  # noqa
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/sdrwacker/miniforge3/envs/jupyter/lib/python3.12/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Users/sdrwacker/miniforge3/envs/jupyter/lib/python3.12/subprocess.py", line 1953, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/sdrwacker/miniconda3/envs/ds/bin/python'


[Errno 2] No such file or directory: '/Users/sdrwacker/miniconda3/envs/ds/bin/python'
[✓] Checking Jupyter engine render....OK

```

**Comment by sorenwacker** [comment]
Interestingly this happens only when I add:

```

```{css}
code {
  font-size: 1.3em;
  /* or try font-size: xx-large; */
}
```
in my file.

**Comment by mcanouil** [comment]
@sorenwacker Please fill completely the issue template. For instance, you did not provide anything to help us reproduce.
Providing only partial snippets of code is not helpful for us to help you.
Also note, that you can edit your first messages to include all the sections you did not fill.

Using `{language}` means computations. If there are no computations, then Conda or else does not matter as they won't be used.

**Comment by cscheid** [comment]
I think @sorenwacker is reporting a failure with `quarto check` directly, so no repro necessary.

@sorenwacker Unfortunately, that means that your Jupyter setup is corrupted somehow. One common source of trouble is the presence of stale Jupyter kernelspec files. That's not something that Quarto can control, but will affect Quarto's ability to use Jupyter effectively.

> Suddenly, when doing quarto preview quarto fails with the error that the conda environment does not exist anymore. It was removed long time ago.

What probably happened is that the removal of the conda environment was incomplete, and left just one such stale Jupyter kernelspec file. Try inspecting the output of

```
$ jupyter kernelspec list
```

In my machine, I get

```
$ jupyter kernelspec list
Available kernels:
  echo         /Users/cscheid/virtualenvs/homebrew-python3/share/jupyter/kernels/echo
  python3      /Users/cscheid/virtualenvs/homebrew-python3/share/jupyter/kernels/python3
  deno         /Users/cscheid/Library/Jupyter/kernels/deno
  julia-1.9    /Users/cscheid/Library/Jupyter/kernels/julia-1.9
```

Each of these directories has a file called `kernel.json`, and this will have an `argv` field:

```
$ cat /Users/cscheid/virtualenvs/homebrew-python3/share/jupyter/kernels/echo/kernel.json
{"argv": ["/Users/cscheid/virtualenvs/homebrew-python3/bin/python3.10", "-m", "echo_kernel", "-f", "{connection_file}"], "display_name": "Echo", "language": "text"}
```

If that argv file is pointing to a missing executable, then we'll run into trouble. Sadly, the inability to reason statically about stale kernels is a misfeature of Jupyter kernels that Quarto can't control, so you'll need to fix it yourself.

**Comment by cderv** [comment]
@sorenwacker is this happening in a terminal or inside an IDE context like VSCODE ? 

Asking because in VSCODE, there could be some configuration impacting what quarto will use and those configuration could be outdated. 

Otherwise, quarto should use the python version found in PATH, and using the `python3` engine (or any other based in computation chunk). It will use the first one found among the registered jupyter engine. 

In your case , it seems it tries to use  `/Users/sdrwacker/miniconda3/envs/ds/bin/python` and I see in the engine list 
````
[✓] Checking Python 3 installation....OK
      Version: 3.12.2 (Conda)
      Path: /Users/sdrwacker/miniforge3/envs/jupyter/bin/python
      Jupyter: 5.7.2
      Kernels: ds, py3, python3
````

So it seems there is still a `ds` Kernels registered for Jupyter. And I wonder if this kernels may be configured to your old envs. Maybe removing the environment did not configured the kernel associated. 

You could try running 
````
/Users/sdrwacker/miniforge3/envs/jupyter/bin/python -m jupyter kernelspec list
````
and see the list of known kernels and look into `kernel.json` of one of the folder to see if there would still be a reference to your old environment. 

Just some ideas based on what you shared. 

Hope it helps

**Comment by cderv** [comment]
Oups. Carlos beat me to it, and posted while I was writing. Same ideas.

**Comment by GaryNapier** [comment]
Hi, I just had the same problem. I changed the `argv` argument in the `Users/<username>/miniforge3/share/jupyter/kernels/python3/kernel.json` file to `/Users/<username>/miniforge3/bin/python`. It was saying `miniconda3` instead of `miniforge3`. 

But why would it do that when I (think) I have erased all traces of Miniconda? I presume there's something left on my system that is making Jupyter think this is the right path, but what could it be and does anyone know how to fix it? 

Thanks
